packages/browser/coretest/__tests__/StitchAppClientIntTests.ts

```diff
@@ -154,6 +159,98 @@ describe("StitchAppClient", () => {
     await client.auth.logout();
     expect(client.auth.isLoggedIn).toBeFalsy();
     expect(client.auth.user).not.toBeDefined();
+
+    // Log back into the last user
+    await client.auth.loginWithCredential(new UserPasswordCredential(
+      "test2@10gen.com", "hunter2"
+    ));
+    expect(client.auth.isLoggedIn).toBeTruthy();
+    expect(client.auth.user!.loggedInProviderType).toEqual(
+      UserPasswordAuthProvider.TYPE
+    );
+
+    expect(client.auth.listUsers().length).toEqual(3);
+    
+    // verify ordering
+    expect(client.auth.listUsers()[0].id).toEqual(anonUser.id);
+    expect(client.auth.listUsers()[1].id).toEqual(emailUserId);
+    expect(client.auth.listUsers()[2].id).toEqual(id2);
+
+    // imitate an app restart
+    Stitch.clearApps();
+    client = harness.getAppClient(appResponse as AppResponse, storage);
+
+    // check everything is as it was
+    expect(client.auth.listUsers().length).toEqual(3);
+    expect(client.auth.isLoggedIn).toBeTruthy();
+    expect(client.auth.user!.loggedInProviderType).toEqual(
+      UserPasswordAuthProvider.TYPE
+    );
+
+    expect(client.auth.listUsers().length).toEqual(3);
+    
+    // verify ordering is preserved
+    expect(client.auth.listUsers()[0].id).toEqual(anonUser.id);
+    expect(client.auth.listUsers()[1].id).toEqual(emailUserId);
+    expect(client.auth.listUsers()[2].id).toEqual(id2);
+
+    // verify that removing the user with id2 removes the second email/pass user
+    // and logs out the active user
+    await client.auth.logoutUserWithId(id2);
```

Comment:
For caution, id add a test whenever you log out to check that the array persisted it: expect(client.auth.listUsers()[2].isLoggedIn).toEqual(false)